src/templates/outstream/porvata-template.js

```diff
@@ -0,0 +1,123 @@
+import { context, events, Porvata, slotTweaker } from '@wikia/ad-engine';
+import * as videoUserInterface from '../interface/video';
+
+export const DEFAULT_VIDEO_ASPECT_RATIO = 640 / 360;
+export const IMA_VPAID_INSECURE_MODE = 2;
+
+export class PorvataTemplate {
+	static getName() {
+		return 'porvata3';
+	}
+
+	static getDefaultConfig() {
+		return {
+			isFloatingEnabled: true,
+			inViewportOffsetTop: 0,
+			inViewportOffsetBottom: 0
+		};
+	}
+
+	constructor(adSlot) {
+		this.adSlot = adSlot;
+		this.config = context.get('templates.porvata3');
+	}
+
+	init(params) {
+		const slotName = this.adSlot.getSlotName();
+		const isInsecureMode = params.vpaidMode === IMA_VPAID_INSECURE_MODE;
+
+		if (!Porvata.isVideoAutoplaySupported()) {
+			return this.adSlot.collapse();
+		}
+
+		if (isInsecureMode) {
+			params.originalContainer = params.container;
+			params.container = this.createVideoContainer(slotName);
+		}
+
+		slotTweaker.collapse(this.adSlot);
+
+		return slotTweaker.makeResponsive(this.adSlot, DEFAULT_VIDEO_ASPECT_RATIO)
+			.then(() => Porvata.inject(params).then(video => this.onReady(video, params)));
+	}
+
+	onReady(video, params) {
+		const slotElement = this.adSlot.getElement();
+		const template = videoUserInterface.selectTemplate(video.videoSettings);
+		const isInsecureMode = params.vpaidMode === IMA_VPAID_INSECURE_MODE;
+		const videoContainer = params.container;
+
+		if (isInsecureMode) {
+			this.adjustVpaidPlayer(video, videoContainer);
+		}
+
+		slotElement.classList.add('porvata-outstream');
+
+		video.addEventListener('loaded', () => {
+			video.container.classList.remove('hide');
+		});
+
+		window.addEventListener('resize', () => {
+			if (!video.isFloating) {
+				const slotWidth = slotElement.clientWidth;
+				video.resize(slotWidth, slotWidth / DEFAULT_VIDEO_ASPECT_RATIO);
+			}
+		});
+
+		this.handleSlotStatus(video);
+
+		events.once(events.PAGE_CHANGE_EVENT, () => {
+			video.destroy();
+		});
+
+		videoUserInterface.setup(video, template, {
+			container: videoContainer,
+			inViewportOffsetTop: this.config.inViewportOffsetTop,
+			inViewportOffsetBottom: this.config.inViewportOffsetBottom,
+			isFloatingEnabled: this.config.isFloatingEnabled && params.enableInContentFloating,
+			slotName: params.slotName
+		});
+
+		return video;
+	}
+
+	handleSlotStatus(video) {
+		video.addEventListener('wikiaAdsManagerLoaded', () => {
+			this.adSlot.success();
+		});
+
+		video.addEventListener('wikiaEmptyAd', () => {
+			this.adSlot.collapse();
+		});
+	}
+
+	adjustVpaidPlayer(video, container) {
+		const videoPlayer = container.querySelector('.video-player');
+
+		video.addEventListener('loaded', () => {
+			const ad = video.ima.getAdsManager().getCurrentAd();
+
+			if (ad && Porvata.isVpaid(ad.getContentType) || '') {
```

Comment:
What is the `|| ''` part of the condition for?

Reply:
Nice catch 💎 It's a typo. Fixed